### ISSUE TITLE
time is wrong when dt modified by a kernel

### ISSUE THREAD

**Issue by delandmeterp** [OceanParcels/parcels#544]
In the main temporal loop, we have
`p.dt = dt`
Then, the kernels are run.
Then we have
`p.time += sign_dt * dt`
Instead, it should be set to 
`p.time += p.dt`
to take into account any modification of `p.dt` in the kernels, no?

**Comment by erikvansebille** [comment]
Can you create a unit test where the `dt` is changed in the kernel, to check if this is indeed a bug?

**Comment by delandmeterp** [comment]
The solution should not be simply using p.dt after kernel, because a modified dt could have various consequences on kernels (e.g. subkernels are using a different dt, t+dt > tend, ...)

So we should check automatically after kernel execution that p.dt has not been modified. If it is the case, we reset the values at before kernel state, and rerun the kernel with new dt (basically, doing automatically what rk45 was doing...)

**Comment by delandmeterp** [comment]
Kernel in `RK45` is like this:
```
    if kappa <= math.fabs(particle.dt * tol[0]):
        particle.lon = lon_4th
        particle.lat = lat_4th
        if kappa <= math.fabs(particle.dt * tol[0] / 10):
            particle.dt *= 2
    else:
        particle.dt /= 2
        return ErrorCode.Repeat
```
So actually, particle.dt can be modified and kernel not repeated.

So far, everything works if particle.dt is changed only at the end of the kernel. Otherwise, what happens has errors

**Comment by erikvansebille** [comment]
Fixed in #657
